src/rpc-handler.test.ts

```diff
@@ -345,15 +345,49 @@ describe('keyringRpcDispatcher', () => {
     );
   });
 
-  it('should fail to list requests with a non-UUIDv4 request ID', async () => {
+  it('calls the keyring with a non-UUIDv4 string request ID', async () => {
     const request: JsonRpcRequest = {
       jsonrpc: '2.0',
-      id: 'invalid-id-string',
+      id: 'request-id',
       method: 'keyring_listRequests',
     };
 
+    keyring.listRequests.mockResolvedValue([]);
+    expect(await handleKeyringRequest(keyring, request)).toStrictEqual([]);
+  });
+
+  it('calls the keyring with a number request ID', async () => {
+    const request: JsonRpcRequest = {
+      jsonrpc: '2.0',
+      id: 1,
+      method: 'keyring_listRequests',
+    };
+
+    keyring.listRequests.mockResolvedValue([]);
+    expect(await handleKeyringRequest(keyring, request)).toStrictEqual([]);
+  });
+
+  it('calls the keyring with a null request ID', async () => {
+    const request: JsonRpcRequest = {
+      jsonrpc: '2.0',
+      id: null,
+      method: 'keyring_listRequests',
+    };
+
+    keyring.listRequests.mockResolvedValue([]);
+    expect(await handleKeyringRequest(keyring, request)).toStrictEqual([]);
+  });
```

Comment:
The specs also state that if the `id` is null then the request is considered a notification, and a response should not be returned. Should we enforce this?

Reply:
It's slightly different. Notifications don't have an ID:
> A Notification is a Request object without an "id" member. [...]

And the ID of a request:
> An identifier established by the Client that MUST contain a String, Number, or NULL value if included. If it is not included it is assumed to be a notification. [...]